src/Analyzers/CSharp/Analyzers/NamingStyle/CSharpNamingStyleDiagnosticAnalyzer.cs

```diff
@@ -27,8 +27,9 @@ internal sealed class CSharpNamingStyleDiagnosticAnalyzer : NamingStyleDiagnosti
         // Parameters of positional record declarations should be ignored because they also
         // considered properties, and that naming style makes more sense
         protected override bool ShouldIgnore(ISymbol symbol)
-            => symbol.IsKind(SymbolKind.Parameter)
-            && IsParameterOfRecordDeclaration(symbol.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax());
+            => (symbol.IsKind(SymbolKind.Parameter)
+            && IsParameterOfRecordDeclaration(symbol.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax()))
+            || !symbol.CanBeReferencedByName;
```

Comment:
this expression is too confusing for me (it mixes `&&` and `||` and `!`).  Can you please make this a normal method with statements.  and then doc each case as to why it should apply.  Thanks!